sky/authentication.py

```diff
@@ -589,6 +593,39 @@ def setup_hyperbolic_authentication(config: Dict[str, Any]) -> Dict[str, Any]:
     # Set up auth section for Ray template
     config.setdefault('auth', {})
     config['auth']['ssh_user'] = 'ubuntu'
-    config['auth']['ssh_public_key'] = public_key_path
 
     return configure_ssh_info(config)
+
+
+def setup_primeintellect_authentication(
+        config: Dict[str, Any]) -> Dict[str, Any]:
+    """Sets up SSH authentication for Prime Intellect.
+    - Generates a new SSH key pair if one does not exist.
+    - Adds the public SSH key to the user's Prime Intellect account.
+    """
+    prime_config = '~/.prime/config.json'
+    if not os.path.isfile(os.path.expanduser(prime_config)):
+        raise Exception(f'{prime_config} does not exist')
```

Comment:
Too broad for me. Let's have something like `FileNotFoundError` or `RuntimeError`.

Reply:
And also we can follow EAFP :)